### DISCUSSION TITLE
let NiceGUI supply HTTPS URL in LAN?

### DISCUSSION THREAD

**Discussion by retsyo** [zauberzeug/nicegui#2190] [Q&A]
### Question

in ["capture user image?"](https://github.com/zauberzeug/nicegui/discussions/2010), I mentioned that the code can't display camera view, instead it shows a blank area on Android/Apple phone, safari on MacOS. After some search, I found some people, for example
- https://stackoverflow.com/a/52500514/4692635 
- https://github.com/jeromeetienne/AR.js/issues/463
say that HTTPS link should be used. I don't know whether its is true or not. 

When NiceGUI appliaction runs, it prompts some available HTTP address. So my question becomes to how to let a NiceGUI, which runs in LAN , offer a HTTPS address?

Thanks

**Comment by rodja** [comment]
NiceGUI is build on top of FastAPI which extensively writes about HTTPS. Please check out https://fastapi.tiangolo.com/de/deployment/https/. In most cases it's simpler to let a reverse proxy like Traefik handle the security for you.
